internal/resource/config/session/settings/session_settings_data_source.go

```diff
@@ -0,0 +1,108 @@
+package sessionsettings
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	client "github.com/pingidentity/pingfederate-go-client/v1125/configurationapi"
+	"github.com/pingidentity/terraform-provider-pingfederate/internal/datasource/common/id"
+	"github.com/pingidentity/terraform-provider-pingfederate/internal/resource/config"
+	internaltypes "github.com/pingidentity/terraform-provider-pingfederate/internal/types"
+)
+
+// Ensure the implementation satisfies the expected interfaces.
+var (
+	_ datasource.DataSource              = &sessionSettingsDataSource{}
+	_ datasource.DataSourceWithConfigure = &sessionSettingsDataSource{}
+)
+
+// NewSessionSettingsResource is a helper function to simplify the provider implementation.
+func NewSessionSettingsDataSource() datasource.DataSource {
+	return &sessionSettingsDataSource{}
+}
+
+// sessionSettingsResource is the resource implementation.
+type sessionSettingsDataSource struct {
+	providerConfig internaltypes.ProviderConfiguration
+	apiClient      *client.APIClient
+}
+
+type sessionSettingsDataSourceModel struct {
+	Id                            types.String `tfsdk:"id"`
+	TrackAdapterSessionsForLogout types.Bool   `tfsdk:"track_adapter_sessions_for_logout"`
+	RevokeUserSessionOnLogout     types.Bool   `tfsdk:"revoke_user_session_on_logout"`
+	SessionRevocationLifetime     types.Int64  `tfsdk:"session_revocation_lifetime"`
+}
+
+// GetSchema defines the schema for the resource.
+func (r *sessionSettingsDataSource) Schema(ctx context.Context, req datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	schema := schema.Schema{
+		Description: "Describes the general session management settings.",
+		Attributes: map[string]schema.Attribute{
+			"track_adapter_sessions_for_logout": schema.BoolAttribute{
+				Description: "Determines whether adapter sessions are tracked for cleanup during single logout. The default is false.",
+				Computed:    true,
+				Optional:    false,
+			},
+			"revoke_user_session_on_logout": schema.BoolAttribute{
+				Description: "Determines whether the user's session is revoked on logout. If this property is not provided on a PUT, the setting is left unchanged.",
```

Comment:
You could remove the language about PUT requests from these descriptions, since this is a data source